include/boost/numeric/odeint/stepper/controlled_adams_bashforth_moulton.hpp

```diff
@@ -39,39 +39,54 @@ class default_order_adjuster
     : m_algebra( algebra )
     {};
 
-    void adjust_order(size_t &order, const boost::array<wrapped_state_type, 3> & xerr)
+    void adjust_order(size_t &order, size_t init, boost::array<wrapped_state_type, 4> &xerr)
```

Comment:
Would it be more clear if this function returned the new order, instead of having an in/out parameter?

Reply:
Not entirely sure if it is more clear, but you are right, it should return the noew order instead of changing it directly.